core/sdk-core/src/main/java/software/amazon/awssdk/core/SelectedAuthScheme.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// TODO: Figure out the right module/package for this. auth? http-auth?
+
+package software.amazon.awssdk.core;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.auth.spi.AuthSchemeOption;
+import software.amazon.awssdk.http.auth.spi.HttpSigner;
+import software.amazon.awssdk.identity.spi.Identity;
+import software.amazon.awssdk.identity.spi.IdentityProvider;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * A container for the identity resolver, signer and auth option that we selected for use with this service call attempt.
+ */
+@SdkProtectedApi
```

Comment:
The code generated auth scheme interceptor needs to instantiate it and set it as attribute, which SigningStage (in core module) reads. Since they are different modules I understood it can't be `@SdkInternalApi`.